examples/gateway/request.py

```diff
@@ -0,0 +1,33 @@
+import requests
+
+
+def main():
+    PREFIX = "http://127.0.0.1:5000/gateway/routes"
+
+    print("Completions")
+    with requests.post(
+        f"{PREFIX}/completions",
+        json={"prompt": "Hello, my name is"},
+    ) as resp:
```

Comment:
I tried the following and got a 500 internal error:


```
    print("Completions")
    with requests.post(
        f"{PREFIX}/completions",
        json={"prompt": "Respond with a random male name. Hello, my name is: ", "flab": 10},
    ) as resp:
        resp.raise_for_status()
        print(resp.json())
```

This is because `flab` isn't a valid param, but I would expect the response to be a 400 bad request and explain that the param is bad

Reply:
Thanks for the catch, I'll add a pedantic model representing the request payload and use it to detect invalid parameters.

Reply:
I also tried

```
    print("Completions")
    with requests.post(
        f"{PREFIX}/completions",
        json={"prompt": "Respond with a random male name. Hello, my name is: ", "temperature": 10},
    ) as resp:
        resp.raise_for_status()
        print(resp.json())
```

which yielded

```
Traceback (most recent call last):
  File "request.py", line 33, in <module>
    main()
  File "request.py", line 12, in main
    resp.raise_for_status()
  File "/Users/corey.zumar/opt/anaconda3/envs/base2/lib/python3.8/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 422 Client Error: Unprocessable Entity for url: http://127.0.0.1:8888/gateway/routes/completions
```

Does OpenAI provide any additional info in the error message about the invalid temperature? If so, can we forward that along?

Reply:
> Thanks for the catch, I'll add a pedantic model representing the request payload and use it to detect invalid parameters.

I don't think we should try to validate every param in the OpenAI spec, since that means our gateway will need to be updated every time OpenAI adds a param / changes the value types. I think we should just forward back the errors gracefully

Reply:
You're right, I'll fix the erorr handling.

Reply:
> Does OpenAI provide any additional info in the error message about the invalid temperature? If so, can we forward that along?

Let me check

Reply:
The response to `temperature=-1` for chat.

```json
{
  "id": "cmpl-7RDPz8zemejQKQSULFYD6lSvfHDOt",
  "object": "text_completion",
  "created": 1686720791,
  "model": "text-davinci-003",
  "choices": [
    {
      "text": "\n\nMy name is Sarah.",
      "index": 0,
      "logprobs": null,
      "finish_reason": "stop"
    }
  ],
  "usage": {
    "prompt_tokens": 5,
    "completion_tokens": 7,
    "total_tokens": 12
  }
}

{
  "error": {
    "message": "-1 is less than the minimum of 0 - 'temperature'",
    "type": "invalid_request_error",
    "param": null,
    "code": null
  }
}
```

Reply:
Made a change. 

```python
    PREFIX = "http://127.0.0.1:5000/gateway/routes"

    print("Completions")
    with requests.post(
        f"{PREFIX}/completions",
        json={"prompt": "Hello, my name is", "hello": "world"},
    ) as resp:
        # resp.raise_for_status()
        print(resp.json())  # -> {'detail': 'Unrecognized request argument supplied: hello'}
```

Reply:
Another example:

```
def main():
    PREFIX = "http://127.0.0.1:5000/gateway/routes"

    print("Completions")
    with requests.post(
        f"{PREFIX}/completions",
        json={"prompt": "Hello, my name is", "temperature": 10},
    ) as resp:
        # resp.raise_for_status()
        print(
            resp.json()
        )  # -> {'detail': [{'loc': ['body', 'temperature'], 'msg': 'ensure this value is less than or equal to 1', 'type': 'value_error.number.not_le', 'ctx': {'limit_value': 1}}]}
```

Reply:
Awesome! Looks great!